Sources/MongoSwift/WriteConcern.swift

```diff
@@ -131,41 +105,76 @@ public class WriteConcern: Codable {
         }
     }
 
-    /// Initializes a new `WriteConcern` by copying a `mongoc_write_concern_t`.
+    /// Initializes a new `WriteConcern` with the same values as the provided `mongoc_write_concern_t`.
     /// The caller is responsible for freeing the original `mongoc_write_concern_t`.
     internal init(from writeConcern: OpaquePointer?) {
-        self._writeConcern = mongoc_write_concern_copy(writeConcern)
-    }
+        if mongoc_write_concern_journal_is_set(writeConcern) {
+            self.journal = mongoc_write_concern_get_journal(writeConcern)
+        } else {
+            self.journal = nil
+        }
+       let number = mongoc_write_concern_get_w(writeConcern)
+        switch number {
+        case MONGOC_WRITE_CONCERN_W_DEFAULT:
+            self.w = nil
+        case MONGOC_WRITE_CONCERN_W_MAJORITY:
+            self.w = .majority
+        case MONGOC_WRITE_CONCERN_W_TAG:
+            if let wTag = mongoc_write_concern_get_wtag(writeConcern) {
+                self.w = .tag(String(cString: wTag))
+            } else {
+                self.w = nil
+            }
+        default:
+            self.w = .number(number)
+        }
 
-    private enum CodingKeys: String, CodingKey {
-        case w, j, wtimeout
+       let wtimeout = Int64(mongoc_write_concern_get_wtimeout(writeConcern))
```

Comment:
I think this line down through the end of the `else` is also misaligned, looks like it all needs to be forward one space.